### ISSUE TITLE
Type checks fail?

### ISSUE THREAD

**Issue by fruitl00p** [kalessil/phpinspectionsea#1540]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | both Php Inspections (EA Extended) and Php Inspections (EA Ultimate), which version |
| Language level | PHP 7+                        |

Current behaviour
--
The inspections don't trigger any type of error / warning ?

Expected behaviour
--
The $a usage should be warned / errored about
---

```php
<?php
declare(strict_types=1);

/**
 * @return null
 */
function test() { return null; }

/**
 * @param \stdClass $b
 */
function test2(\stdClass $b){ }
$a = test();
test2($a);
```

The usage of $a in the test2() function should give an error as we specifically want (i.e.) an \stdClass which isnt declared as nullable

**Comment by fruitl00p** [comment]
adding to this: changing the `@return null` to `@return int` *does* warn...

**Comment by fruitl00p** [comment]
Another update: this seems to be related to the callee (test2 in this case) not accepting a nullable argument...

**Comment by kalessil** [comment]
@fruitl00p, which inspection you are referencing to (perhaps it's possible to share an example error message)? I'm guessing we are talking about an bundled IDE inspection, but want to ensure.

**Comment by fruitl00p** [comment]
@kalessil it might be a non existant check since i would have expected it to error out on one of the insepections but it doesnt... (some sort of type check should fail... ? right?)

Or does this go beyond and should i look for something like phpstan for this type of inference?

**Comment by kalessil** [comment]
It recalls like some strict types checking from PhpStorm as well (PHP | PHP Strict Standards or PHP | Type compatibility). But I personally would go with phpstan/psalm. For me personally psalm causes less hassle with a lot of value, but phpstan is also a great addition.

**Comment by kalessil** [comment]
I'll close the issue as third party, so I can focus on our scope better.
